EDMarketConnector.py

```diff
@@ -925,56 +934,89 @@ def getandsend(self, event=None, retrying: bool = False):  # noqa: C901, CCR001
             self.button['state'] = self.theme_button['state'] = tk.DISABLED
             self.w.update_idletasks()
 
+        query_time = int(time())
+        logger.trace_if('capi.worker', 'Requesting full station data')
+        config.set('querytime', query_time)
+        companion.session.station(
+            query_time=query_time, tk_response_event=self._CAPI_RESPONSE_TK_EVENT_NAME,
+            play_sound=play_sound
+        )
+
+    def capi_handle_response(self, event=None):  # noqa: C901, CCR001
+        """Handle the resulting data from a CAPI query."""
+        play_bad: bool = False
+        err: Optional[str] = None
+
+        capi_response: Union[companion.EDMCCAPIFailedRequest, companion.EDMCCAPIResponse]
         try:
-            querytime = int(time())
-            data = companion.session.station()
-            config.set('querytime', querytime)
+            logger.trace_if('capi.worker', 'Pulling answer off queue')
+            capi_response = companion.session.capi_response_queue.get(block=False)
+            if isinstance(capi_response, companion.EDMCCAPIFailedRequest):
+                logger.trace_if('capi.worker', f'Failed Request: {capi_response.message}')
+                if capi_response.exception:
+                    raise capi_response.exception
+
+                else:
+                    raise ValueError(capi_response.message)
+
+            logger.trace_if('capi.worker', 'Answer is not a Failure')
+            if not isinstance(capi_response, companion.EDMCCAPIResponse):
+                msg = f"Response was neither CAPIFailedRequest nor EDMCAPIResponse: {type(capi_response)}"
```

Comment:
Quote style

```suggestion
                msg = f'Response was neither CAPIFailedRequest nor EDMCAPIResponse: {type(capi_response)}'
```

Reply:
Probably a consequence of original wording have an apostrophe in it.  Changed.